### PR TITLE
Some cosmetic changes

### PR DESCRIPTION
**Pull request by xvw**
A first patch that separates my first small modifications so that the patch that truncates the stacktraces isn't too big.

### CODE REVIEW

File: .ocamlformat [hunk 1]
```diff
@@ -1,2 +1 @@
-version=0.24.1
```
Comment (pitag-ha) [inline .ocamlformat:1]: Why have you removed the version?

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 Sonja Heinze
+
```
Comment (pitag-ha) [inline LICENSE:4]: Thanks! I should have added a license months ago. I'd like to add 3Rafal here as well. We can leave it like this for now, and I ask Rafal when he's back if I can add him.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Merl-an
+
+> Tool for [merlin](https://github.com/ocaml/merlin) performance analysis
+
+Merl-an generates queries to test [merlin](https://github.com/ocaml/merlin)
+performance, executes them and gathers the results.
```
Comment (pitag-ha) [inline README.md:6]:
```suggestion
Merl-an generates queries to analyze [merlin](https://github.com/ocaml/merlin)
performance and behavior, executes them and gathers the results.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+## Local development
+
+Since **merl-an** streamlines Merlin's main trunk, we recommend building a local
+switch and manually adding the dependency to Merlin:
+
+```shell
+# opam update
+# opam switch create . ocaml-base-compiler.4.14.1 --deps-only -y
```
Comment (pitag-ha) [inline README.md:15]: Which compiler version is needed will depend on the state of merlin's main branch. So probably the best is to create an empty switch and have the compiler version be solved by opam.

File: dune-project [hunk 5]
```diff
@@ -30,4 +30,6 @@
   ppx_enumerate
   ppx_fields_conv
   (ppx_yojson_conv (<= v0.15.1))
-  ptime))
+  ptime
+  (ocamlformat :with-dev-setup)
+  (ocp-indent :with-dev-setup)))
```
Comment (pitag-ha) [inline dune-project:35]: Why do you want `ocp-indent` as a dev dependency?

About `ocamlformat`: If we add it as a dependency, we need to specify its version. In that case, in the future, we'd need to update both the ocamlformat and the dune-project file when bumping `ocamlformat`. I don't have a strong opinion about this. If you want to add `ocamlformat` with version as dev dependency, that's fine for me.

File: lib/merlin.ml [hunk 6]
```diff
@@ -247,53 +242,47 @@ module Cmd = struct
...
-               merlin
-               (Query_type.to_string query_type)
-               (Location.print_edge Right)
-               loc File.pp file File.pp file
+          let+ loc = retrieve_loc ~query_type loc in
+          Format.asprintf " %a %s -look-for ml -position '%a' -filename %a < %a"
+            basic_cmd merlin
+            (Query_type.to_string query_type)
+            (Location.print_edge Right)
+            loc File.pp file File.pp file
```
Comment (pitag-ha) [inline lib/merlin.ml:250]: Lol. Good catch! xD

File: test/behavior.t/run.t [hunk 7]
```diff
@@ -18,15 +18,15 @@
...
-  {"sample_id":1,"cmd":"ocamlmerlin server type-enclosing -position '3:12' -index 0 -filename test.ml < test.ml","responses":[{"class":"return","value":[{"start":{"line":3,"col":12},"end":{"line":3,"col":13},"type":"int -> int -> int","tail":"no"},{"start":{"line":3,"col":10},"end":{"line":3,"col":15},"type":1,"tail":"no"},{"start":{"line":3,"col":6},"end":{"line":3,"col":15},"type":2,"tail":"no"}],"notifications":[],"query_num":1}]}
-  {"sample_id":0,"cmd":"ocamlmerlin server case-analysis -start '3:14' -end '3:14' -filename test.ml < test.ml","responses":[{"class":"return","value":[{"start":{"line":3,"col":14},"end":{"line":3,"col":15}},"(match 3 with | 0 -> _ | _ -> _)"],"notifications":[],"query_num":0}]}
+  {"sample_id":12,"cmd":" ocamlmerlin server locate -look-for ml -position '3:12' -filename test1.ml < test1.ml","responses":[{"class":"return","value":{"file":"stdlib.ml","pos":{"line":94,"col":0}},"notifications":[],"heap_mbytes":4,"cache":{"reader_phase":"miss","ppx_phase":"miss","typer":{"reused":1,"typed":1},"cmt":{"hit":2,"miss":0},"cmi":{"hit":1,"miss":0}},"query_num":12}]}
```
Comment (pitag-ha) [inline test/behavior.t/run.t:21]: I'd filter `heap_mbytes` out, potentially also the cache hits and misses component. The `heap_mbytes`captures how much Merlin allocates during the query. That's not deterministic.

**Review by pitag-ha — COMMENTED** [review]
Thanks! Every clean-up pass is appreciated!

I think you've unintentionally added a few empty files as well. Or are they on purpose?

File: .ocamlformat (hunk 1 above)
Comment (xvw) [reply .ocamlformat:1]: Having a fixed version is good for formatting consistency, but implies having to fix the OCamlformat version at dependency level.

So there are two solutions:
- fix OCamlformat in the dune file
- let any version of OCamlformat work (which is the case here)

I can do what you want.

File: README.md (hunk 4 above)
Comment (xvw) [reply README.md:15]: Since adding merlin `pinned` happens after Switch initialization, I'm not sure if it works.

File: lib/merlin.ml (hunk 6 above)
Comment (xvw) [reply lib/merlin.ml:250]: Yes, in many case, I still prefer the combo `let*` + `return` if there is more than one case but here I guess that it is only for result deconstruction so I prefer remove a level of indentation.

File: test/behavior.t/run.t (hunk 7 above)
Comment (xvw) [reply test/behavior.t/run.t:21]: Maybe I can leave it to the current PR i am currently working on?

File: .ocamlformat (hunk 1 above)
Comment (pitag-ha) [reply .ocamlformat:1]: If we don't fix the version, every developer would have different formatting, so most PRs would contain formatting changes as well, right?

Comment (xvw) [reply .ocamlformat:1]: Yes

File: README.md (hunk 4 above)
Comment (pitag-ha) [reply README.md:15]: I think it should work. You can create an empty switch via
```
opam switch create --empty <switch-name>
```
(or a local switch, if you prefer that)
Afterwards, you pin merlin, which will than automatically also install the correct OCaml version into the switch as it isn't installed yet:
```
opam pin https://github.com/ocaml/merlin.git
```

Let me know if I've misunderstood you!

File: dune-project (hunk 5 above)
Comment (pitag-ha) [reply dune-project:35]: I'm curious: Why do you still use ocp-indent while also using ocamlformat?

In any case, you can rely on ocp-indent! I'm just wondering why you'd impose it on others by making it a dev dependency.

File: test/behavior.t/run.t (hunk 7 above)
Comment (pitag-ha) [reply test/behavior.t/run.t:21]: Sure! In that case, I'd remove the test file modifications in this PR.

(This repo isn't at all kept very clean, though, so I won't be stubborn on it. If you prefer to merge this with unclean tests and fix the tests later, let's just do that.)

File: .ocamlformat (hunk 1 above)
Comment (pitag-ha) [reply .ocamlformat:1]: And you think that's better than synching the ocamlformat version between all devs?

**Review by pitag-ha — APPROVED** [review]
Looks all good. Thank you!

(I'm still surprised that there weren't any formatting changes when bumping ocamlformat from 0.24 to 0.26. Maybe, lately ocamlformat changes less than it used to.)

File: merl-an.opam [hunk 8]
```diff
@@ -21,6 +21,11 @@ depends: [
   "ppx_fields_conv"
   "ppx_yojson_conv" {<= "v0.15.1"}
   "ptime"
+  "merlin-lib"
+  "dot-merlin-reader"
+  "merlin"
```
Comment (pitag-ha) [inline merl-an.opam:26]: Could you make merlin a test dependency (i.e. `with-test`)? Having merlin installed is not needed to compile `merl-an`.

**Comment by pitag-ha** [comment]
Nice, thanks for the clean-ups and fixes!

**Pull request merged**